src/components/select/select.template.tsx

```diff
@@ -305,17 +308,30 @@ function getSelectedIcon(value: any) {
         return '';
     }
 
+    const name = getIconName(value.icon);
+    const color = getIconColor(value.icon, value.iconColor);
     const style: any = {};
-    if (value.iconColor) {
-        style.color = value.iconColor;
+    if (color) {
+        style.color = color;
     }
 
     return (
         <limel-icon
             class="limel-select__selected-option__icon"
-            name={value.icon}
+            name={name}
             size="medium"
             style={style}
         />
     );
 }
+
+export function triggerIconColorWarning(options: Option[]) {
+    options.forEach((option) => {
+        if (option.iconColor) {
+            /* eslint-disable-next-line no-console */
+            console.warn(
+                "The `iconColor` prop is deprecated now! Use the new `Icon` interface and instead of `iconColor: 'color-name'` write `icon {name: 'icon-name', color: 'color-name'}`."
+            );
+        }
+    });
+}
```

Comment:
This should generate console.warn for limel-select **ONLY** when `iconColor` is used. But the warning in the console constantly shows up and I have no idea how to fix this. I tried a lot.

Reply:
@jgroth do you have any magic here? I can't figure out what the problem is here, and why the console warning is always being displayed.

Reply:
https://github.com/Lundalogik/lime-elements/pull/2542/files#r1377545428

Reply:
Ah I see 😓 tnx